src/components/address-input/_macro.spec.js

```diff
@@ -87,126 +45,209 @@ describe('macro: address-input', () => {
-                error: { text: 'Server error: town or city' },
+        });
+        describe('WHEN: Params: line1', () => {
+            test('THEN: it renders "address line 1" input field with expected parameters', () => {
+                expect(inputSpy.occurrences).toContainEqual({
+                    id: 'address-input-example-id-line1',
+                    name: 'address-input-example-id-line1',
+                    classes: 'ons-js-address-line1',
+                    label: {
+                        text: 'Address line 1',
+                    },
+                    value: 'Flat 12345',
+                    width: '20@m',
+                    error: { text: 'Server error: address line 1' },
+                });
             });
         });
 
-        it('renders "postcode" input field with expected parameters', () => {
-            const faker = templateFaker();
-            const inputSpy = faker.spy('input', { suppressOutput: true });
+        describe('WHEN: Params: line2', () => {
+            test('THEN: it renders "address line 2" input field with expected parameters', () => {
+                expect(inputSpy.occurrences).toContainEqual({
+                    id: 'address-input-example-id-line2',
+                    name: 'address-input-example-id-line2',
+                    classes: 'ons-js-address-line2',
+                    label: {
+                        text: 'Address line 2',
+                    },
+                    value: '12345 The Road',
+                    width: '20@m',
+                    error: { text: 'Server error: address line 2' },
+                });
+            });
+        });
 
-            faker.renderComponent('address-input', {
-                ...EXAMPLE_AUTOSUGGEST_ADDRESS_MINIMAL,
-                ...EXAMPLE_MANUAL_INPUT_FIELDS,
-                isEditable: true,
+        describe('WHEN: Params: town', () => {
+            test('THEN: it renders "town or city" input field with expected parameters', () => {
+                expect(inputSpy.occurrences).toContainEqual({
+                    id: 'address-input-example-id-town',
+                    name: 'address-input-example-id-town',
+                    classes: 'ons-js-address-town',
+                    label: {
+                        text: 'Town or city',
+                    },
+                    value: 'The Town',
+                    error: { text: 'Server error: town or city' },
+                });
             });
+        });
 
-            expect(inputSpy.occurrences).toContainEqual({
-                id: 'address-input-example-id-postcode',
-                name: 'address-input-example-id-postcode',
-                classes: 'ons-js-address-postcode',
-                label: {
-                    text: 'Postcode',
-                },
-                value: 'PO57 6ODE',
-                width: '7',
-                error: { text: 'Server error: postcode' },
+        describe('WHEN: Params: postcode', () => {
```

Comment:
For these whens here they should be worded differently as they are worded more like a given. For example this one would be something like "WHEN: postcode is provided" and its the same with all the other parts of the address